dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java

```diff
@@ -495,33 +521,32 @@ public final Response logoutAs(@Context final HttpServletRequest httpServletRequ
 	 *
 	 * @return The list of users that can be impersonated.
 	 *
-	 * @deprecated use {@link com.dotcms.rest.api.v2.user.UserResource#loginAsData(HttpServletRequest, String, int, int)}
+	 * @deprecated use {@link com.dotcms.rest.api.v2.user.UserResource#loginAsData(HttpServletRequest, HttpServletResponse, String, int, int)}
 	 */
 	@GET
 	@Path("/loginAsData")
 	@JSONP
 	@NoCache
 	@Deprecated
 	@Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
-	public final Response loginAsData(@Context final HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @QueryParam("filter") String filter,
-			@QueryParam("includeUsersCount") boolean includeUsersCount) {
-		Response response = null;
+	public final Response loginAsData(@Context final HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @QueryParam("filter") final String filter,
+			@QueryParam("includeUsersCount") final boolean includeUsersCount) {
+		Response response;
+		User currentUser = new User();
 		try {
-			InitDataObject initData = //webResource.init(null, httpServletRequest, httpServletResponse, true, null);
-
-			new WebResource.InitBuilder(webResource)
+			final InitDataObject initData = new WebResource.InitBuilder(webResource)
 					.requiredBackendUser(true)
 					.requiredFrontendUser(false)
 					.requestAndResponse(httpServletRequest, httpServletResponse)
 					.rejectWhenNoUser(true)
 					.init();
 
-			User currentUser = initData.getUser();
-
+			currentUser = initData.getUser();
 			response = Response.ok(this.helper.getLoginAsUsers(currentUser, filter, includeUsersCount)).build();
-		} catch (Exception e) {
-			SecurityLogger.logInfo(UserResource.class, "An error occurred when processing the request. " + e.getMessage());
-			response = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+		} catch (final Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5162183)